src/dataflow-types/src/types.rs

```diff
@@ -751,12 +751,31 @@ pub mod sources {
         User(String),
     }
 
+    /// `SourceEnvelope`s, describe how to turn a stream of messages from `SourceConnector`s,
+    /// and turn them into a _differential stream_, that is, a stream of (data, time, diff)
+    /// triples.
+    ///
+    /// Some sources (namely postgres and pubnub) skip any explicit envelope handling, effectively
+    /// asserting that `SourceEnvelope` is `None` with `KeyEnvelope::None`.
+    // TODO(guswynn): update this ^ when SimpleSource is gone.
     #[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
     pub enum SourceEnvelope {
-        /// If present, include the key columns as an output column of the source with the given properties.
+        /// The most trivial version is `None`, which typically produces triples where the diff
+        /// is ALWAYS `1`
+        /// TODO(guswynn): update `None` docs to describe `-1` diff cases
```

Comment:
Can you say more about the `-1` diff case? Seems at odds with the prior assertion that this always produces diffs of `1`?

Reply:
this is just a reminder for me to fix this when I make `SourceEnvelope::None` support `-1` for pg deletes

I can clarify this comment!